src/Mvc/Mvc.Core/src/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -143,6 +143,11 @@ public async Task BindModelAsync(ModelBindingContext bindingContext)
 
             if (formatter == null)
             {
+                if (AllowEmptyBody && httpContext.Request.ContentLength == 0)
```

Comment:
@Tratcher isn't there a better way to check for empty bodies?

Reply:
https://github.com/dotnet/aspnetcore/blob/55bb14c972f9d273ed5c35275e0314df63c4c8b2/src/Http/Http.Features/src/IHttpRequestBodyDetectionFeature.cs#L27